src/test/fuzz/inputfetcher.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <inputfetcher.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/transaction_identifier.h>
+
+#include <cstdint>
+
+FUZZ_TARGET(inputfetcher)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    const auto batch_size{
+        fuzzed_data_provider.ConsumeIntegralInRange<int32_t>(0, 1024)};
+    const auto worker_threads{
+        fuzzed_data_provider.ConsumeIntegralInRange<int32_t>(0, 15)};
+    InputFetcher fetcher{batch_size, worker_threads};
+
+    CCoinsView dummy;
+    CCoinsViewCache db(&dummy);
+    CCoinsViewCache cache(&db);
+
+    CBlock block;
+    Txid prevhash{Txid::FromUint256(ConsumeUInt256(fuzzed_data_provider))};
+
+    const auto txs{fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1,
+        std::numeric_limits<uint32_t>::max())};
+    for (uint32_t i{0}; i < txs; ++i) {
```

Comment:
This will create very long running inputs (e.g. txs = std::numeric_limits<uint32_t>::max()).

```suggestion
    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), N) {
```

or

```suggestion
    LIMITED_WHILE(fuzzed_data_provider.remaining_bytes(), N) {
```

Reply:
Thanks, done!